src/pages/Rooms/Rooms.js

```diff
@@ -0,0 +1,126 @@
+  grid-auto-flow: dense;
+
+`;
+const Item = styled.div`
+  justify-self: stretch;
+`;
+
+class Rooms extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {};
+  }
+
+  componentDidMount() {
+    // Option 1) start a polling request to get the state
+    // this.pollingTimer = setInterval(() => this.fetchState(), 1000);
+
+    // Option 2) open a websocket and receive incremental state updates
+    this.fetchState(); // fetch the initial state
+    //TODO handle ws lifecycle management
+    this.ws = new WebSocket(
+      `${window.location.protocol === 'https:' ? 'wss' : 'ws'}://${window.location.host}/websocket`
+    );
+    this.ws.onopen = () => console.log('websocket connected');
+    this.ws.onclose = () => console.log('websocket closed');
+    this.ws.onmessage = message => {
+      let event;
+      try {
+        event = JSON.parse(message.data);
+        console.log('ws received', event);
+      } catch (err) {
+        console.log('ws received message is not a json: ', message.data);
+      }
+      if (event && event.type === 'stateChange') {
+        this.setState(state => setWith(state, `${event.path}.${event.id}`, event.value, clone));
+      }
+    };
+  }
+
+  componentWillUnmount() {
+    clearInterval(this.pollingTimer);
+    this.ws.close();
+  }
+
+  fetchState() {
+    fetch(`/API/state`)
+      .then(res => res.json())
+      .then(result => this.setState(result))
+      .catch(error => console.error('Error:', error));
+  }
+
+  handleButtonClick(id) {
+    var event = {
+      type: 'buttonClicked',
+      id
+    };
+    // Option 1) Post an event to the API
+    // this.postEvent(event);
+
+    // Option 2) Post an event to the websocket
+    console.log('ws send', event);
+    this.ws.send(JSON.stringify(event));
+  }
+
+  postEvent(event) {
+    fetch(`/API/event`, {
+      method: 'POST', // or 'PUT'
+      body: JSON.stringify(event), // data can be `string` or {object}!
+      headers: {
+        'Content-Type': 'application/json'
+      }
+    })
+      .then(res => res.json())
+      .then(result => this.setState(result))
+      .catch(error => console.error('Error:', error));
+  }
+  //implement changes to any of the UI elements
+  notifyChange(changeInfo) {
+    console.log(JSON.stringify(changeInfo));
+  }
+  render() {
+    const renderCards = () => {
+      var order = 1;
+      const cards = [];
+      //Rooms
+      const roomsArr = Object.values(Object.values(testHome)[0]);
+      roomsArr.map(room => {
+        const r = (
+          <Item key={order}>
+            <Room
+              toggles={room.toggles}
+              name={room.name}
+              notifyChange={changeInfo => this.notifyChange(changeInfo)}
+            ></Room>
+          </Item>
+        );
+        console.log('Rooms ordernumber= ' + order);
+        cards.push(r);
+        order++;
+      });
+
+      return cards;
+    };
+    return (
+      <div>
+        <Container>{renderCards()}</Container>
```

Comment:
cant use var or const in the return of the render function: "Expression expected"

Reply:
Don't quite get it in this example key still is the array index since the map (and forEach) according to MDN returns:
The index of the current element being processed in the array.

Reply:
note the [Object.entries](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/entries)!

Reply:
note the [Object.entries](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/entries)!

Reply:
but the argument key used in the `.map(([key, room]) => (`  is according to MDN just the array index of the current element so it doesnt solve the fact that an array index is used as key:
https://github.com/RoetsNv/ui/pull/1#discussion_r307998832

Reply:
nope, look carefully (or debug if you don't believe me ;)).
As you can read in [Object.entries](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/entries) it returns an array of [key, value] pairs of an object. So an array of arrays. (use debugger to check what Object.entries(testHome.rooms) looks like!).
that array will be mapped with the given arrow function, that only take 1 argument, according to [Array.map](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/map) the first argument will be the current value of the array. Which will be another array with a [key, value] pair of the testHome.rooms object. That array is already [destructered](http://es6-features.org/#ParameterContextMatching) in the arrow function arg list.